### PR TITLE
ConDec-53: Add jira.view.issue context to decision knowledge management resources

### PR DESCRIPTION
**Pull request by kleebaum**
If we don't provide this context (or even the context atl.general) some methods (e.g. buildTreant) cannot be found in the JIRA issue view.

### CODE REVIEW

**Comment by kleebaum** [comment]
`	/* TODO:
	   What is the intention of the below 4 lines?
	   Should the thread wait until callDialogFromView is done?
	   If yes, it can be done better than below.
	 */
	callDialogFromView();
	closeDialog();
	callDialogFromView();
	closeDialog();
	callDialogFromView();`
@lw2011 , yes the thread should wait until callDialogFromView is done. Do you have a solution proposal for this?

**Comment by lw2011** [comment]
Hi @kleebaum 
 > @lw2011 , yes the thread should wait until callDialogFromView is done. Do you have a solution proposal for this?

I assume the problems that were mitigated here where caused by the slow and asynchronous DOM manipulations by the callDialogFromView function. If the assumption is correct, there are several solutions to this, some would be:

* The callDialogFromView could return a Promise, so it guarantees that when it is done, the rest of the code can continue.
* The callDialogFromView could receive a callback as argument and execute it when it is done.

Cheers
Lukasz

File: src/main/resources/js/jstree/jstree.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
         // TODO: remove DEBUG lines
         console.log("jstree.js init");
         console.log("$ ver");
-        console.log($.fn.jquery);
+        //console.log($.fn.jquery); // $ is not found because we add <context>jira.view.issue</context> in the atlassian-plugin.xml
```
Comment (lw2011) [inline src/main/resources/js/jstree/jstree.js]: This line and the line above should be just removed. The source code comment `// $ is not found ...` could be moved to the commit message.

File: src/main/resources/js/jstree/jstree.js [hunk 2]
```diff
@@ -26,7 +26,7 @@
     console.log($.fn.jquery);
     
     console.log("window.$ ver");
-    console.log(window.$.fn.jquery);
+    //console.log(window.$.fn.jquery); // $ is not found because we add <context>jira.view.issue</context> in the atlassian-plugin.xml
```
Comment (lw2011) [inline src/main/resources/js/jstree/jstree.js]: same as above:
This line and the line above should be just removed. The source code comment `// $ is not found ...` could be moved to the commit message.

File: src/main/resources/js/view.context.menu.js [hunk 3]
```diff
@@ -67,18 +66,19 @@ function setUpDialogForCreateAction(id) {
...
-		e.preventDefault();
-		AJS.dialog2("#dialog").hide();
+	document.getElementById("dialog-cancel-button").addEventListener("click", function() {
+		closeDialog();
 	});
+	AJS.dialog2("#dialog").show();
+	AJS.dialog2("#dialog").hide();
+	AJS.dialog2("#dialog").show();
+	AJS.dialog2("#dialog").hide();
+	AJS.dialog2("#dialog").show();
```
Comment (kleebaum) [inline src/main/resources/js/view.context.menu.js]: @lw2011 and @Jcl15 Unfortunately, I could not manage to get rid of this call sequence using callback functions or promise. I tried the following but this does not work:
`$.when(setUpDialogForCreateAction(id)).done(function() {
            AJS.dialog2("#dialog").show();
        });`
Could you please try to solve this in code and commit it?

**Pull request merged**